### PR TITLE
feat: Clinical soft filter

### PR DESCRIPTION
**Pull request by fevac**
## Description

[PR description]

#### Added

- [Description]

#### Changed

- [Description]

#### Fixed

- [Description]

#### Removed

- [Description]

## Documentation

- [ ] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

### Feature Tests

- [ ] N/A
- [ ] Test [Description]
    - [Screenshot]

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [x] N/A
    - [ ] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### Panel of Normal specific criteria

- [ ] The PR includes the addition of a new Panel of Normals
- [ ] The samples have been verified to adhere to the sample selection criteria on [Atlas PoN creation instructions for Balsamic](https://atlas.scilifelab.se/infrastructure/BALSAMIC/balsamic_pon/#procedure)

### User Changes

- [x] N/A
- [ ] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [ ] Affected users have been included in the development process and given a chance to provide feedback.

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [x] N/A
    - [ ] Updated: [Link]
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

### Validation criteria

Validation criteria to be added to validation report PR: [LINK-TO-VALIDATION-REPORT-PR from the [validations repository](https://github.com/Clinical-Genomics/validations/)]

**Version specific criteria**

- Text here or N/A

> [!IMPORTANT]
> One of the below checkboxes for validation need to be checked 

- [ ] Added version specific validation criteria to validation report
- [ ] Changes validated in standard sections: [validation-section]
- [ ] Validation criteria not necessary

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [ ] Provided a comprehensive description of the PR.
    - [ ] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [ ] Verified and updated documentation if necessary.
- **Validation criteria**
    - [ ] Completed the validation criteria section of the template.
- **Tests**
    - [ ] Described and tested the functionality addressed in the PR.
    - [ ] Ensured integration of the new code with existing workflows.
    - [ ] Confirmed that meaningful unit tests were added for the changes introduced.
    - [ ] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [ ] Addressed and resolved all the feedback provided during the code review process.
    - [ ] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [ ] Code implements the intended features or fixes the reported issue.
    - [ ] Code follows the project's coding standards and style guide.
- **Documentation**
    - [ ] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Validation criteria**
    - [ ] The author has completed the validation criteria section of the template
- **Tests**
    - [ ] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [ ] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

File: BALSAMIC/constants/cache.py [hunk 1]
```diff
@@ -334,21 +348,21 @@ class DockerContainers(StrEnum):
...
-            "url": "https://raw.githubusercontent.com/etal/cnvkit/master/data/access-5k-mappable.hg19.bed",
+        "clinical_genes": {
+            "url": "ADDURL",
             "file_type": FileType.TXT,
             "gzip": False,
-            "file_name": "access_5kb_hg38.txt",
+            "file_name": "clinical_genes_v0.1",
             "dir_name": "genome",
         },
-        "delly_exclusion": {
```
Comment (mathiasbio) [inline BALSAMIC/constants/cache.py:347]: Why is this removed?

File: BALSAMIC/constants/cache.py [hunk 2]
```diff
@@ -160,6 +160,20 @@ class DockerContainers(StrEnum):
             "file_name": "cancer_rank_model_-v0.2-.ini",
             "dir_name": "genome",
         },
+        "clinical_genes": {
+            "url": "ADDURL",
```
Comment (mathiasbio) [inline BALSAMIC/constants/cache.py:164]: Here you are planning to add the link to https://raw.githubusercontent.com/Clinical-Genomics/reference-files ? :)

File: BALSAMIC/constants/cache.py [hunk 3]
```diff
@@ -160,6 +160,20 @@ class DockerContainers(StrEnum):
...
+            "file_type": FileType.TXT,
+            "gzip": False,
+            "file_name": "clinical_genes_v0.1",
+            "dir_name": "genome",
+        },
+        "clinical_fusions": {
+            "url": "ADDURL",
```
Comment (mathiasbio) [inline BALSAMIC/constants/cache.py:171]: Here you are planning to add the link to https://raw.githubusercontent.com/Clinical-Genomics/reference-files ? :)

File: BALSAMIC/assets/scripts/clinical_annotation.py [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    include_fusion_partner_info(record, parsed_csq_entries)
+    include_clinical_info(record, annotation)
+    soft_filter_off_panel(record)
+    clean_record_info_fields(record)
+    return record
+
+
+def update_header(reader) -> None:
+    # Add info fields to header
+    headers = [
```
Comment (mathiasbio) [inline BALSAMIC/assets/scripts/clinical_annotation.py:229]: Have you checked to see if any of these new INFO fields clashes with any of the existing ones?

File: BALSAMIC/snakemake_rules/annotation/varcaller_sv_filter.rule [hunk 5]
```diff
@@ -59,3 +58,27 @@ tabix -p vcf -f {output.vcf_pass_svdb};
 
 bcftools +counts {output.vcf_pass_svdb} > {output.bcftools_counts};
     """
+
+rule clinical_soft_filter_sv:
```
Comment (mathiasbio) [inline BALSAMIC/snakemake_rules/annotation/varcaller_sv_filter.rule:62]: Is this rule applied for all workflows or just WGS? Have we considered if it should be applied to all workflows or only some?
